File: knavwid/src/main/java/com/github/programmerr47/knavwid/CustomLayoutNavigationBuilder.java
package com.github.programmerr47.knavwid;

import com.github.programmerr47.knavwid.layoutfactory.LayoutFactory;

public final class CustomLayoutNavigationBuilder extends NavigationBuilder<CustomLayoutNavigationBuilder> {

    public CustomLayoutNavigationBuilder(LayoutFactory layoutFactory) {
        super(layoutFactory, NavigationDefaults.NavigationDefaultsHolder.navigationDefaults());
    }

    @Override
    protected CustomLayoutNavigationBuilder getThis() {
        return this;
    }

    public CustomLayoutNavigationBuilder toolbarId(int toolbarId) {
        this.toolbarId = toolbarId;
        return this;
    }

    public CustomLayoutNavigationBuilder bottomBarId(int bottomBarId) {
        this.bottomBarId = bottomBarId;
        return this;
    }

    public AutoLayoutNavigationBuilder auto() {
        return new AutoLayoutNavigationBuilder(layoutFactory());
    }
}

File: knavwid/src/main/java/com/github/programmerr47/knavwid/CustomLayoutNavigationBuilder.kt
package com.github.programmerr47.knavwid

import com.github.programmerr47.knavwid.layoutfactory.LayoutFactory

class CustomLayoutNavigationBuilder(layoutFactory: LayoutFactory) : NavigationBuilder<CustomLayoutNavigationBuilder>(
    layoutFactory,
    NavigationDefaults.NavigationDefaultsHolder.navigationDefaults()!!
) {

    override val `this`: CustomLayoutNavigationBuilder get() = this

    fun toolbarId(toolbarId: Int) = apply {
        this.toolbarId = toolbarId
    }

    fun bottomBarId(bottomBarId: Int) = apply {
        this.bottomBarId = bottomBarId
    }

    fun auto() = AutoLayoutNavigationBuilder(layoutFactory())
}
